drivers/nrf_wifi/fw_if/umac_if/src/fmac_util.c

```diff
@@ -368,6 +369,73 @@ bool nrf_wifi_util_is_arr_zero(unsigned char *arr,
 	return true;
 }
 
+#if defined(NRF70_PROMISC_DATA_RX)
+bool nrf_wifi_util_check_filt_setting(struct nrf_wifi_fmac_vif_ctx *vif,
+				      unsigned short *frame_control)
```

Comment:
Since this is not a `generic` utility function I would propose having them in a separate file specific to `promiscuous` mode.